### PR TITLE
Added support for max_pool_connections on s3 driver

### PR DESCRIPTION
**Pull request by leandroreox**
Signed-off-by: Leandro Reox <leitan@MacBook-Pro-de-Leandro.local>

### CODE REVIEW

File: gnocchi/storage/s3.py [hunk 1]
```diff
@@ -50,6 +50,12 @@
                  help="Maximum time to wait checking data consistency when "
                  "writing to S3. Set to 0 to disable data consistency "
                  "validation."),
+    cfg.FloatOpt('s3_max_pool_connections',
```
Comment (chungg) [inline gnocchi/storage/s3.py]: should probably be IntOpt

File: gnocchi/storage/s3.py [hunk 2]
```diff
@@ -50,6 +50,12 @@
...
+                 min=10,
+                 default=50,
+                 help="The maximum number of connections to keep in a "
+                 "connection pool. If this value is not set, the default "
+                 "value of 10 is used."),
```
Comment (chungg) [inline gnocchi/storage/s3.py]: i'm not sure the last sentence is correct. can you unset the value if there is a default?

Comment (leandroreox) [reply gnocchi/storage/s3.py]: no, ill change that, i extracted it from the boto doc

File: gnocchi/storage/s3.py (hunk 1 above)
Comment (leandroreox) [reply gnocchi/storage/s3.py]: yes, changing that

**Comment by chungg** [comment]
could you squash all the fixes into a single commit? something like this: https://www.devroom.io/2011/07/05/git-squash-your-latests-commits-into-one/

and then when you push, use git push -f origin master

**Comment by leandroreox** [comment]
Merged al changes into a single squashed commit

File: gnocchi/storage/s3.py [hunk 3]
```diff
@@ -50,6 +50,11 @@
                  help="Maximum time to wait checking data consistency when "
                  "writing to S3. Set to 0 to disable data consistency "
                  "validation."),
+    cfg.IntOpt('s3_max_pool_connections',
+                min=10,
```
Comment (jd) [inline gnocchi/storage/s3.py]: I this really 10? Seems to me that 1 should work.

Comment (leandroreox) [reply gnocchi/storage/s3.py]: @jd  10 is de default value that comes on the adapter.py of the python requests library, i really think lowering it is not a good call: https://github.com/requests/requests/blob/master/requests/adapters.py#L47

Comment (jd) [reply gnocchi/storage/s3.py]: It's not saying it's a good idea. It's saying it's possible. That's what this min/max values need to represent.

The _default_ value is what should be a good idea. :)

**Comment by leandroreox** [comment]
Changed the minimun value since what @jd said makes a lot of sense

**Comment by jd** [comment]
You still need to fix the tests it seems.

**Comment by leandroreox** [comment]
@jd squashed :)

File: requirements.txt [hunk 4]
```diff
@@ -23,3 +23,4 @@ Paste
 PasteDeploy
 monotonic
 daiquiri
+pyparsing>=2.2.0
```
Comment (jd) [inline requirements.txt]: uh?

Comment (jd) [reply requirements.txt]: You probably need to rebase on master

**Comment by jd** [comment]
@leandroreox I suggest using https://github.com/jd/git-pull-request :)

**Comment by leandroreox** [comment]
@jd merged from upstream

**Comment by jd** [comment]
@leandroreox any update/fix?

**Comment by leandroreox** [comment]
@jd what im seeing is that the test are failing with "boto3 unavailable" thats seems to be triggered when boto3 cannot be imported, boto3 is there since day one, i didnt touched that part, am i missing something here ? or this tests should not be passing like ever ?

**Comment by leandroreox** [comment]
@jd fixed

**Comment by chungg** [comment]
you should squash your commits :)

File: gnocchi/common/s3.py [hunk 5]
```diff
@@ -41,7 +42,9 @@ def get_connection(conf):
         endpoint_url=conf.s3_endpoint_url,
         region_name=conf.s3_region_name,
         aws_access_key_id=conf.s3_access_key_id,
-        aws_secret_access_key=conf.s3_secret_access_key)
+        aws_secret_access_key=conf.s3_secret_access_key,
+        config = boto_config.Config(max_pool_connections=conf.
+                 s3_max_pool_connections))
```
Comment (chungg) [inline gnocchi/common/s3.py]:
```
        config = boto_config.Config(
            max_pool_connections=conf.s3_max_pool_connections))
```

**Comment by leandroreox** [comment]
Yes @chungg  i have done it kind that way, but commit and squash was pending, this should pass al the tests

**Comment by jd** [comment]
@leandroreox looks good but you need to squash those commits

**Comment by jd** [comment]
@leandroreox do you need help to rebase/squash?

**Comment by leandroreox** [comment]
@jd yes, now i have merged with master i have a lot of commits in between mine, im looking how to rebase without screwing the other commits

**Comment by chungg** [comment]
oh, you pushed your change to your master :/ ... i don't know what the correct way to fix that is. usually it's best (i think) to create:

1) sync your local master with upstream master
2) create a new branch
3) push that branch

maybe just reset your fork and push a new PR? the others know git better than me

**Comment by jd** [comment]
@leandroreox You want to do `git rebase -i master` on your branch and put everything except the "main" commit on `squash` so they're all squashed. Then write a proper commit message and should should only have one commit. Resend your branch (use `git pull-request` if you're lazy) and you should be done.

**Comment by leandroreox** [comment]
@jd this also applies if my branch is master ?

**Comment by jd** [comment]
@leandroreox yeah that might not work, so you need to precise the master branch from gnocchi, which is very likely `origin/master`, so do `git rebase --interactive origin/master`

**Comment by leandroreox** [comment]
Sorry about the whole rebase stuff, still learning how not to get fuc*** by git, is some action from my side still neede ? i saw that @jd commited the PR. Thanks

**Comment by jd** [comment]
@leandroreox don't worry, I took care, you're good :)

**Comment by leandroreox** [comment]
@jd haha thanks, promise to do the next better

**Pull request merged**